tests/test_tomography_gate_set_tomography.py

```diff
@@ -0,0 +1,392 @@
+    nqubits = 2
+    test_circuit = qibo.models.Circuit(nqubits)
+    test_circuit.add(gates.H(0))
+    test_circuit.add(gates.S(0))
+    test_circuit.add(gates.RX(1, np.pi / 3))
+
+    inverse_circuit = reset_register(test_circuit, (1,))
+
+    correct_gates = [[gates.RX(1, np.pi / 3).dagger()]]
+
+    for groundtruth, gate in zip(correct_gates, inverse_circuit.queue):
+        assert isinstance(gate, type(groundtruth[0]))
+
+
+def test_reset_register_2():
+    # Test resetting both qubits
+    nqubits = 2
+    test_circuit = qibo.models.Circuit(nqubits)
+    test_circuit.add(gates.H(0))
+    test_circuit.add(gates.CNOT(0, 1))
+
+    inverse_circuit = reset_register(test_circuit, (0, 1))
+
+    correct_gates = [
+        [gates.CNOT(0, 1)],
+        [gates.H(0)],
+    ]
+    for groundtruth, gate in zip(correct_gates, inverse_circuit.queue):
+        assert isinstance(gate, type(groundtruth[0]))
+
+
+@pytest.mark.parametrize("a, b", [(0, 2), (1, 2), (2, 3)])
+def test_reset_register_invalid_tuple(a, b):
+    # Test resetting both qubits
+
+    nqubits = 2
+    test_circuit = qibo.models.Circuit(nqubits)
+    test_circuit.add(gates.H(0))
+    test_circuit.add(gates.CNOT(0, 1))
+
+    # Check if NameError is raised
+    with pytest.raises(NameError):
+        inverse_circuit = reset_register(test_circuit, (a, b))
+
+
+def test_GST(backend):
+    T = np.array([[1, 1, 1, 1], [0, 0, 1, 0], [0, 0, 0, 1], [1, -1, 0, 0]])
+    target_gates = [gates.SX(0), gates.Z(0), gates.CY(0, 1)]
+    target_matrices = [g.matrix() for g in target_gates]
+    # superoperator representation of the target gates in the pauli basis
+    target_matrices = [to_pauli_liouville(m, normalize=True) for m in target_matrices]
+    gate_set = [g.__class__ for g in target_gates]
+    lam = 1e-2
+    depol = NoiseModel()
+    depol.add(DepolarizingError(lam))
+    empty_1q, empty_2q, *approx_gates = GST(
+        gate_set,
+        nshots=int(1e4),
+        noise_model=depol,
+        include_empty=True,
+        backend=backend,
+    )
+    for target, estimate in zip(target_matrices, approx_gates):
+        transf = empty_1q if estimate.shape[0] == 4 else empty_2q
+        T_matrix = T if estimate.shape[0] == 4 else np.kron(T, T)
+        estimated_matrix = (
+            T_matrix @ np.linalg.inv(transf) @ estimate @ np.linalg.inv(T_matrix)
+        )
+        backend.assert_allclose(
+            target,
+            estimated_matrix,
+            atol=1e-1,
+        )
+
+
+def test_GST_invalid_gate(backend):
+    T = np.array([[1, 1, 1, 1], [0, 0, 1, 0], [0, 0, 0, 1], [1, -1, 0, 0]])
+    target_gates = [gates.TOFFOLI(0, 1, 2)]
+    target_matrices = [g.matrix() for g in target_gates]
+    # superoperator representation of the target gates in the pauli basis
+    target_matrices = [to_pauli_liouville(m, normalize=True) for m in target_matrices]
+    gate_set = [g.__class__ for g in target_gates]
+    lam = 1e-2
+    depol = NoiseModel()
+    depol.add(DepolarizingError(lam))
```

Comment:
I don't think any of these is needed as you are only testing for the error to be raised

Reply:
What do you mean? Do we not include the depolarizing noise?

Reply:
You are just testing that an error is raised when you try to perform `GST` on a TOFFOLI, you don't need to define a noise model and the target matrices, the error is gonna be raised regardless. Actually, the best thing would be to merge the three `GST` tests in a single one using the pytest parameters as I suggested above.